components/server/src/workspace/workspace-starter.ts

```diff
@@ -1251,6 +1251,11 @@ export class WorkspaceStarter {
             ideImage = ideConfig.ideOptions.options[ideConfig.ideOptions.defaultIde].image;
         }
 
+        let workspaceClass: string = "";
+        if (await this.userService.userGetsMoreResources(user)) {
+            workspaceClass = "gitpod.io/internal/xl";
```

Comment:
We cannot use that as a node label

`May 10 00:13:11 headless-ws-ephemeral86-pool-internal-xl-vp2f k3s[2536]: E0510 00:13:11.540005    2536 kubelet_node_status.go:92] "Unable to register node with API server" err="Node \"headless-ws-ephemeral86-pool-internal-xl-vp2f\" is invalid: metadata.labels: Invalid value: \"gitpod.io/internal/xl\": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')" node="headless-ws-ephemeral86-pool-internal-xl-vp2f"`